packages/core/src/modal/modal.element.ts

```diff
@@ -137,7 +137,7 @@ export class CdsModal extends CdsInternalOverlay {
 
   disconnectedCallback() {
     super.disconnectedCallback();
-    this.observers.forEach(o => o.disconnect());
+    this.observers.forEach(o => o?.disconnect());
```

Comment:
For context with anyone reviewing this, the `?` is needed because there is a rare case where a component can be added to the dom and `connectedCallback` fires and then immediately be removed and cause `disconnectedCallback` to fire. When this happens lifecycle hooks like `firstUpdated` will not fire and the `observer` may be not yet set or undefined.

The use case where this came up was when running the react/angular unit tests. The elements are added checked to exist and then immediately removed.

Reply:
Was that this issue? https://github.com/vmware/clarity/issues/6764

I found that jsdom was throwing an exception that caused the observer to be null or undefined: https://github.com/vmware-clarity/core/pull/7/files#diff-44ab5968ecca127b3fb35008d3463acfd24277d35870f90ebed70850ddb4df26R17

If the type for observers isn't nullable, it might be good to add a comment here so the `?` doesn't get removed in the future since it looks like it's not necessary according to typescript.